### ISSUE TITLE
Ulid.NewUlid() crashes on Galaxy A13 and IL2CPP

### ISSUE THREAD

**Issue by andrei-kopysov** [Cysharp/Ulid#81]
Unity3D 2022.3.13f1
When I make an apk build and run it on Galaxy A13 it crashes on Ulid.NewUlid()
It crashes every time

Stacktrace
```          
Crashed: Thread: SIGBUS  0x00000000ac9fbcea
#00 pc 0x48dd824 libil2cpp.so (Ulid._ctor) (BuildId: cf773a2dd8bd74ee)
#01 pc 0x5429ca2 libil2cpp.so (__Il2CppComDelegate_Finalize_mC9F8EA94444C3AF0A43CC723A23EE4D8B7984F8C) (BuildId: cf773a2dd8bd74ee)
#02 pc 0x542dbbe libil2cpp.so (__Il2CppComDelegate_Finalize_mC9F8EA94444C3AF0A43CC723A23EE4D8B7984F8C) (BuildId: cf773a2dd8bd74ee)
#03 pc 0xbc272 libc.so (BuildId: f1f4a4b2b76c986f7bd6e840119e41e2)
#04 pc 0xbc272 libc.so (BuildId: f1f4a4b2b76c986f7bd6e840119e41e2)
#05 pc 0x48e2186 libil2cpp.so (Ulid._cctor) (BuildId: cf773a2dd8bd74ee)
#06 pc 0x1347ce2 libil2cpp.so (RuntimeInvoker_FalseVector4(void (*)(), MethodInfo const*, void*, void**, void*)) (BuildId: cf773a2dd8bd74ee)
#07 pc 0x14a9b76 libil2cpp.so (il2cpp::vm::Runtime::InvokeWithThrow(MethodInfo const*, void*, void**)) (BuildId: cf773a2dd8bd74ee)
#08 pc 0x9a205c libil2cpp.so (BuildId: cf773a2dd8bd74ee)
#09 pc 0xbc272 libc.so (BuildId: f1f4a4b2b76c986f7bd6e840119e41e2)
#10 pc 0x14a9ab6 libil2cpp.so (il2cpp::vm::Runtime::Invoke(MethodInfo const*, void*, void**, Il2CppException**)) (BuildId: cf773a2dd8bd74ee)
#11 pc 0x56021fa libil2cpp.so (__Il2CppComDelegate_Finalize_mC9F8EA94444C3AF0A43CC723A23EE4D8B7984F8C) (BuildId: cf773a2dd8bd74ee)
#12 pc 0x14a791e libil2cpp.so (il2cpp::vm::Runtime::ClassInit(Il2CppClass*)) (BuildId: cf773a2dd8bd74ee)
#13 pc 0x55756bd libil2cpp.so (__Il2CppComDelegate_Finalize_mC9F8EA94444C3AF0A43CC723A23EE4D8B7984F8C) (BuildId: cf773a2dd8bd74ee)
#14 pc 0x54383c6 libil2cpp.so (__Il2CppComDelegate_Finalize_mC9F8EA94444C3AF0A43CC723A23EE4D8B7984F8C) (BuildId: cf773a2dd8bd74ee)
...
```

SystemInfo CPU = ARMv7 VFPv3 NEON, Cores = 8, Memory = 3707mb
SystemInfo ARM big.LITTLE configuration: 8 big (mask: 0xff), 0 little (mask: 0x0)
Version '2022.3.13f1 (5f90a5ebde0f)', Build type 'Development', Scripting Backend 'il2cpp', CPU 'armeabi-v7a', Stripping 'Enabled'

**Comment by neuecc** [comment]
Is the Galaxy A13 the only Android device that doesn't work?
It would be difficult to test that in my environment...

**Comment by andrei-kopysov** [comment]
@neuecc Based on Crashlytics all these devices below are affected also. Thank you for the support and for your great libraries.

Galaxy J4
Galaxy A02s
Galaxy A03 Core
Galaxy A10
Galaxy A10e
Galaxy A10s
Galaxy A11
Galaxy A13
Galaxy A13 5G
Galaxy M10
Galaxy M13
Galaxy Tab A7 Lite
Poco F1
Poco X4 Pro 5G
Redmi A2
Redmi A2+
Redmi A3
Redmi 9A
Redmi 9C
Redmi 10A
Redmi 10A Sport
Redmi Note 12
Redmi Note 12 5G
Realme Narzo 50i
RMX3231
Realme C30
Realme C30s
Smrat 8 PRO

**Comment by andrei-kopysov** [comment]
I tested on some devices myself and can prove that
It crashes on Galaxy A13 every time
It does not crash on Galaxy Tab A7 Lite 
Build crashes on Redmi 9A and Redmi 9C NFC every time
Build doesn't crash on Redmi Note 12 Pro

I used very simple app for testing - just a new empty Unity project and Ulid.NewUlid() call on Awake.

**Comment by neuecc** [comment]
Thank you.
It seems to occur frequently with MediaTek CPUs.
There don't seem to be any completely similar cases, but there are multiple crash reports for libil2cpp.so.
https://discussions.unity.com/search?q=SIGBUS%20order%3Alatest_topic

How about this fix?
https://issuetracker.unity3d.com/issues/android-application-crash-when-methods-are-called-in-multiple-threads
It seems that this issue was fixed in 2022.3.20f1.
I'd like you to try it out.

**Comment by neuecc** [comment]
There seems to be a problem with 32-bit ARM CPUs.
Come to think of it, I remember having the same problem with the MessagePack library before!
https://github.com/MessagePack-CSharp/MessagePack-CSharp/issues/931

`MemoryMarshal.Cast<byte, ulong>` causes problem.
In Ulid, using similar pattern in construcotr `ref Unsafe.As<long, byte>`.

A method for efficiently avoiding this has been implemented in recent versions of MessagePack.
I'll try a similar approach.

**Comment by neuecc** [comment]
I've released v1.3.4, it avoids Unsafe. As cast so maybe solved this issue.
Please try it.
If that doesn't work, I'll think of a different method (I think that if I just handle it as a pointer in the end, it will work without a problem).

**Comment by andrei-kopysov** [comment]
@neuecc Awesome! 1.3.4 doesn't crash anymore. You fixed that issue. :smiley:
